cubeb-sys/systest/build.rs

```diff
@@ -0,0 +1,38 @@
+extern crate ctest;
+
+use std::env;
+use std::path::PathBuf;
+
+fn main() {
+    let root = PathBuf::from(env::var_os("DEP_CUBEB_ROOT").unwrap());
+
+    let mut cfg = ctest::TestGenerator::new();
+
+    // Include the header files where the C APIs are defined
+    cfg.header("cubeb.h")
+        .header("cubeb_mixer.h")
+        .header("cubeb_panner.h")
+        .header("cubeb_resampler.h");
+
+    // Include the directory where the header files are defined
+    cfg.include(root.join("include"))
+        .include(root.join("include/cubeb"))
+        .include("../libcubeb/src");
+
+    cfg.type_name(|s, _| s.to_string())
+        .field_name(|_, f| match f {
+            "device_type" => "type".to_string(),
+            _ => f.to_string(),
+        });
+
+    // Don't perform signededness checks for callbacks
+    cfg.skip_signededness(|s| match s {
+        s if s.ends_with("_callback") => true,
```

Comment:
What's this for?

Reply:
This tells `ctest` not to check signedness for any type that ends in `_callback`. This will match all the function pointer types for callbacks which are unsigned and fixes a compilation problem.

Reply:
Right, I guess I was curious about what the failure was that required this.

Reply:
It disables checks that:
```c
  ((cubeb_data_callback) -1) < 0)
```
because pointers are always >= 0.

Eg:
```
running: "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-m64" "-I" "/home/djg/Mozilla/cubeb-rs/target/debug/build/cubeb-sys-d44c18bd2119471e/out/include" "-I" "/home/djg/Mozilla/cubeb-rs/target/debug/build/cubeb-sys-d44c18bd2119471e/out/include/cubeb" "-I" "../libcubeb/src" "-Wall" "-Wextra" "-Wall" "-Wextra" "-Werror" "-Wno-unused-parameter" "-Wno-type-limits" "-o" "/home/djg/Mozilla/cubeb-rs/target/debug/build/systest-6cb33ba3fcc20560/out/all.o" "-c" "/home/djg/Mozilla/cubeb-rs/target/debug/build/systest-6cb33ba3fcc20560/out/all.c"
cargo:warning=/home/djg/Mozilla/cubeb-rs/target/debug/build/systest-6cb33ba3fcc20560/out/all.c: In function ‘__test_signed_cubeb_data_callback’:
cargo:warning=/home/djg/Mozilla/cubeb-rs/target/debug/build/systest-6cb33ba3fcc20560/out/all.c:13:52: error: ordered comparison of pointer with integer zero [-Werror=extra]
cargo:warning=                 return (((cubeb_data_callback) -1) < 0);
```

I'll add a comment.